src/dnvm/Channel.cs

```diff
@@ -103,13 +113,24 @@ public static Channel FromString(string str)
             case "preview": return new Preview();
             default:
                 var components = str.Split('.');
-                if (components.Length != 2)
+                switch (components)
                 {
-                    throw new DeserializeException($"Invalid channel version: {str}");
+                    case [_, _]:
+                        var major = int.Parse(components[0]);
+                        var minor = int.Parse(components[1]);
+                        return new VersionedMajorMinor(major, minor);
+                    case [_, _, _]:
+                        if (components[2] is not [<= '9' and >= '0', 'x', 'x'])
+                        {
+                            throw new DeserializeException($"Feature band must be 3 characters and end in 'xx': {str}");
```

Comment:
If this granularity isn't needed, I can inline the pattern. I just figured it may be useful.

Reply:
Nope I like it. Validate during parse, don't worry about it again.